src/open_inwoner/utils/tests/test_migrations.py

```diff
@@ -37,18 +35,71 @@ def setUp(self):
         # Reverse to the original migration
         executor.migrate(self.migrate_from)
 
-        self.setUpBeforeMigration(old_apps)
-
+    def apply_migration_to(self):
         # Run the migration to test
         executor = MigrationExecutor(connection)
         executor.loader.build_graph()  # reload.
         executor.migrate(self.migrate_to)
+        self.apps = executor.loader.project_state(self.migrate_to).apps
 
+    def restore_constraint_checks(self):
         # Restore constraint checks
         with connection.cursor() as cursor:
             cursor.execute("SET CONSTRAINTS ALL DEFERRED")
 
-        self.apps = executor.loader.project_state(self.migrate_to).apps
-
     def setUpBeforeMigration(self, apps):
         pass
+
+
+class TestSuccessfulMigrations(TestMigrationsBase):
+    """Test a successful migration.
+
+    Set the class attributes `app`, `migration_from`, and `migrate_to`. You
+    can specify your pre-migration state in `setUpBeforeMigration()`, and
+    consequently assert against the resulting state (in your test, do
+    not import models directly but use:
+
+    ```
+    MyModel = self.apps.get_model(self.app, "MyModel")
+    ```
+    """
+
+    def setUp(self):
+        self.revert_to_migrate_from()
+        self.setUpBeforeMigration(self.old_apps)
+        self.apply_migration_to()
+        self.restore_constraint_checks()
```

Comment:
The level of abstraction strikes me as uneven across the class. Restoration of constraint checks is abstracted in the method `restore_constraint_checks`, but the original change (`"SET CONSTRAINTS ALL IMMEDIATE"`) is implicit in `setUpBeforeMigration`. Loooking just at this code, I can see immediately that constraints are being restored, but I don't see why that's happening because the original change is hidden. If possible (perhaps there are complications), I'd prefer having both of these implicit or both explicit in a separate method.